### PR TITLE
gh-95132: Correctly relay *args and **kwds from sqlite3.connect to factory

### LINKED ISSUE #95132: Python3.11: sqlite3.connect passes more arguments to factory()
Hello,

**Bug report**

Consider the following code snippet:

$ cat /tmp/test.py
```python
import sqlite3


class SqliteConnection(sqlite3.Connection):
    def __init__(self, *args, **kwargs):
        print(args)
        print(kwargs)
        super(SqliteConnection, self).__init__(*args, **kwargs)


db_path = '/tmp/foo.db'
sqlite3.connect(db_path, factory=SqliteConnection)
```

And the execution:

```bash
$ python3.10 /tmp/test.py
('/tmp/foo.db',)
{'factory': <class '__main__.SqliteConnection'>}

$ python3.11 /tmp/test.py
('/tmp/foo.db', 5.0, 0, '', 1, <class '__main__.SqliteConnection'>, 128, 0)
{}
```
The default values of sqlite3.connect's arguments are passed to the factory in
Python3.11, which was not the case in Python3.10. This means that code like
this will fail:

$ cat /tmp/failure.py
```python
import sqlite3


class SqliteConnection(sqlite3.Connection):
    def __init__(self, *args, **kwargs):
        kwargs['timeout'] = 42
        super(SqliteConnection, self).__init__(*args, **kwargs)


db_path = '/tmp/foo.db'
sqlite3.connect(db_path, factory=SqliteConnection)
```

```bash
$ python3.10 /tmp/failure.py
$

$ python3.11 /tmp/failure.py
Traceback (most recent call last):
  File "/tmp/failure.py", line 11, in <module>
    sqlite3.connect(db_path, factory=SqliteConnection)
  File "/tmp/failure.py", line 7, in __init__
    super(SqliteConnection, self).__init__(*args, **kwargs)
TypeError: Connection() takes at most 8 arguments (9 given)
```

**Your environment**
This behaviour appeared in 185ecdc1463c527743eeb16a5deef75c1985de79 and affects
Python 3.11 and later.

It was mentioned in https://github.com/python/cpython/issues/93044 (see the
second bullet point in the reporter's message) but has not been fixed.

### PR DESCRIPTION
**Pull request by erlend-aasland**
This PR partially reverts gh-24421 (PR) and fixes the remaining concerns
given in gh-93044 (issue).

Fixes gh-95132

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Triggering the bots to be 100% sure we don't break 3.11 again.

**Comment by erlend-aasland** [comment]
I've queued a commit that further improves the factory tests. I'll wait until the bots are done before pushing it, though 🤖

**Comment by erlend-aasland** [comment]
> I've queued a commit that further improves the factory tests. I'll wait until the bots are done before pushing it, though 🤖

Bot run: 87 successful and 1 skipped checks.

File: Modules/_sqlite/module.c [hunk 1]
```diff
@@ -42,47 +42,44 @@ module _sqlite3
...
-
-Opens a connection to the SQLite database file database.
-
-You can use ":memory:" to open a database connection to a database that resides
-in RAM instead of on disk.
-[clinic start generated code]*/
+// NB: This needs to be in sync with the Connection.__init__ docstring.
+PyDoc_STRVAR(module_connect_doc,
+"connect($module, /, database, timeout=5.0, detect_types=0,\n"
+"        isolation_level=<unrepresentable>, check_same_thread=True,\n"
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/module.c]: Why not `isolation_level=''` or `isolation_level='DEFERRED'`?

**Review by serhiy-storchaka — APPROVED** [review]
LGTM, but `<unrepresentable>` makes the exact signature almost useless. And there is no need to use it, because the default value can be represented.

**Comment by erlend-aasland** [comment]
@serhiy-storchaka, would you mind reviewing this? I'd like to land this pretty quick, since there's a new beta around the corner, and, at this point in time in the dev cycle, I prefer it reviewed by another core dev to make sure I didn't miss a spot.

**Comment by erlend-aasland** [comment]
> LGTM, but `<unrepresentable>` makes the exact signature almost useless. And there is no need to use it, because the default value can be represented.

Thanks for reviewing! Yeah, that's a C&P artefact from the clinic generated docstring. I'll fix it. Thanks again!

File: Modules/_sqlite/module.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_sqlite/module.c]:
```suggestion
"        isolation_level="", check_same_thread=True,\n"
```

Comment (serhiy-storchaka) [reply Modules/_sqlite/module.c]: `""` will be eaten by the C compiler.

**Comment by erlend-aasland** [comment]
This PR also addresses the second concern in #93044:

> 2. Keyword arguments are passed as positional arguments to `factory()`.

**Comment by serhiy-storchaka** [comment]
Both the second and the third concerns.

File: Modules/_sqlite/module.c [hunk 2]
```diff
@@ -42,47 +42,44 @@ module _sqlite3
...
-
-Opens a connection to the SQLite database file database.
-
-You can use ":memory:" to open a database connection to a database that resides
-in RAM instead of on disk.
-[clinic start generated code]*/
+// NB: This needs to be in sync with the Connection.__init__ docstring.
+PyDoc_STRVAR(module_connect_doc,
+"connect($module, /, database, timeout=5.0, detect_types=0,\n"
+"        isolation_level="", check_same_thread=True,\n"
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/module.c]:
```suggestion
"        isolation_level='', check_same_thread=True,\n"
```

**Comment by erlend-aasland** [comment]
> Both the second and the third concerns.

Yes, I've already updated my post. Also thanks for the last got'cha.

**Pull request merged**